frontend/catalyst/__init__.py

```diff
@@ -76,6 +76,7 @@
 )
 from catalyst.utils.exceptions import CompileError
 
+autograph_force_fallbacks = False
```

Comment:
I want it exclusively for testing. The idea is to set this variable and test that the fallback parts of autograph work.

Reply:
Interesting. These global flags are going to be removed in the near future to be replaced with a qjit parameter for the autograph mode, with 3 options:
- strict: no fallbacks allowed, if there is an array during the conversion the error is immediately raised to the user
- permissive (default): autograph will always fall back to Python if it can't convert the user code, but in some instances it will raise a warning to allow the user to adjust their code to be more optimal
- silent: same as permissive, but no warnings are raised

I'm not sure how your new flag would fit into this 🤔 @josh146 any ideas?

Reply:
We could have `autograph_mode="force-fallback"`? I'm agnostic here -- as long as there are places where this would be useful, and there are no other ways to solve @grwlf's testing problem, it could be worth supporting.

Reply:
@grwlf Could your testing be solved differently?

Reply:
Hmm, I think the current version is kind of less invasive than others I see (like, triggering an artificial error forcing autograph to fallback), I will think.

Reply:
So I moved the test flag out from the `__init__` into the `ag_primitives` and changed its semantics. @dime10 could you please check?

Reply:
@grwlf thinking about this overnight -- if this is something that is primarily for development testing, and doesn't have as much value to be user-facing, then it implies it might be best as a custom [test fixture](https://docs.pytest.org/en/6.2.x/fixture.html).

That is, a test fixture that sets up a patch to force this behaviour, and then has a tear down. You could include this fixture on the TestClass that requires this behaviour.

Reply:
@josh146 this `monkeypatch` thing that we are using [here](https://github.com/PennyLaneAI/catalyst/blob/6544da88f9d28be42a94a079f8969c204ee9f9fa/frontend/test/pytest/test_autograph.py#L1103) is a fixture, the built-in one. The problem is how to link it to our main code. As I see, Autograph requires us to use functions and we want to trigger the fallback path in them, so we must either use effects (here - the global variable I added) or pass arguments triggering errors on the default path. I think that it would be easier to maintain the former approach, but I am open to suggestions.